### PR TITLE
Use AuthenticatedGcpServiceAccount from requireAuthenticatedAdmin + fake token authentication.

### PR DESCRIPTION
**Pull request by isoos**
- #6140
- Updated audit model entities to handle more cases of generic agents, and also adjusted integrity checks for them.
- Consent handling is still User-based, we are still creating `User` entity in Datastore. Filed #6187 to track its migration.
- Added fake token authentication for tests. Registering is restricted for local use only.
- Updated admin tests with new token edge cases.

### CODE REVIEW

**Review by jonasfj — APPROVED** [review]

File: app/lib/account/agent.dart [hunk 1]
```diff
@@ -72,6 +72,9 @@ abstract class AuthenticatedAgent {
   ///  * For a service account we display a description.
   ///  * For automated publishing we display the service and the origin trigger.
   String get displayId;
+
+  /// The email address of the agent (if there is any).
+  String? get email;
```
[the remark this answers is not in the document]
Comment (isoos) [reply app/lib/account/agent.dart:77]: It makes the construction of audit log records much easier, where we specifically want to store the email field (if available).

File: app/lib/admin/backend.dart [hunk 2]
```diff
@@ -148,7 +149,7 @@ class AdminBackend {
     if (user == null) return;
     if (user.isDeleted) return;
 
-    _logger.info('${caller.userId} (${caller.email}) initiated the delete '
+    _logger.info('${caller.oauthUserId} (${caller.email}) initiated the delete '
```
[the remark this answers is not in the document]
Comment (isoos) [reply app/lib/admin/backend.dart]: Done.

File: app/lib/account/agent.dart [hunk 3]
```diff
@@ -126,6 +132,11 @@ class AuthenticatedGcpServiceAccount implements AuthenticatedAgent {
     required this.idToken,
     required this.payload,
   });
+
+  @override
+  String get email => payload.email;
+  String get oauthUserId => payload.sub;
```
[the remark this answers is not in the document]
Comment (isoos) [reply app/lib/account/agent.dart:138]: Yeah, adding a TODO entry for it.

File: app/lib/fake/backend/fake_auth_provider.dart [hunk 4]
```diff
@@ -66,3 +70,50 @@ String createFakeAuthTokenForEmail(
       path: email.replaceAll('.', '-dot-').replaceAll('@', '-at-'),
       queryParameters: {'aud': audience ?? 'fake-site-audience'}).toString();
 }
+
+String createFakeServiceAccountToken(
+    {required String email, String? audience}) {
+  return Uri(path: 'gcp-service-account', queryParameters: {
+    'email': email,
+    'aud': audience ?? 'https://pub.dev',
+  }).toString();
```
[the remark this answers is not in the document]
Comment (isoos) [reply app/lib/fake/backend/fake_auth_provider.dart:80]: Hm, yeah. I'll try that in a follow-up...
